code/game/atoms.dm

```diff
@@ -726,78 +723,65 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
+/obj/item/clothing/gloves/add_blood(list/blood_dna, b_color)
 	. = ..()
 	transfer_blood = rand(2, 4)
 
-/turf/add_blood(list/blood_dna, color)
+/turf/add_blood(list/blood_dna, b_color)
 	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
 	if(!B)
 		B = new /obj/effect/decal/cleanable/blood/splatter(src)
 	B.transfer_blood_dna(blood_dna) //give blood info to the blood decal.
-	B.basecolor = color
+	B.basecolor = b_color
 	return TRUE //we bloodied the floor
 
-/mob/living/carbon/human/add_blood(list/blood_dna, color)
+/mob/living/carbon/human/add_blood(list/blood_dna, b_color)
 	if(wear_suit)
-		wear_suit.add_blood(blood_dna, color)
-		wear_suit.blood_color = color
+		wear_suit.add_blood(blood_dna, b_color)
 		update_inv_wear_suit()
 	else if(w_uniform)
-		w_uniform.add_blood(blood_dna, color)
-		w_uniform.blood_color = color
+		w_uniform.add_blood(blood_dna, b_color)
 		update_inv_w_uniform()
 	if(head)
-		head.add_blood(blood_dna, color)
-		head.blood_color = color
+		head.add_blood(blood_dna, b_color)
 		update_inv_head()
 	if(glasses)
-		glasses.add_blood(blood_dna, color)
-		glasses.blood_color = color
+		glasses.add_blood(blood_dna, b_color)
 		update_inv_glasses()
 	if(gloves)
 		var/obj/item/clothing/gloves/G = gloves
-		G.add_blood(blood_dna, color)
-		G.blood_color = color
+		G.add_blood(blood_dna, b_color)
 		verbs += /mob/living/carbon/human/proc/bloody_doodle
 	else
-		hand_blood_color = color
+		hand_blood_color = b_color
 		bloody_hands = rand(2, 4)
 		transfer_blood_dna(blood_dna)
 		verbs += /mob/living/carbon/human/proc/bloody_doodle
 
 	update_inv_gloves()	//handles bloody hands overlays and updating
 	return TRUE
 
-/obj/item/proc/add_blood_overlay(color)
+/obj/item/proc/add_blood_overlay()
 	if(initial(icon) && initial(icon_state))
-		//try to find a pre-processed blood-splatter. otherwise, make a new one
-		var/index = blood_splatter_index()
-		var/icon/blood_splatter_icon = GLOB.blood_splatter_icons[index]
-		if(!blood_splatter_icon)
-			blood_splatter_icon = icon(initial(icon), initial(icon_state), frame = 1)		//we only want to apply blood-splatters to the initial icon_state for each object
-			blood_splatter_icon.Blend("#ffffff", ICON_ADD) 			//fills the icon_state with white (except where it's transparent)
-			blood_splatter_icon.Blend(icon('icons/effects/blood.dmi', "itemblood"), ICON_MULTIPLY) //adds blood and the remaining white areas become transparant
-			blood_splatter_icon = fcopy_rsc(blood_splatter_icon)
-			GLOB.blood_splatter_icons[index] = blood_splatter_icon
-
-		blood_overlay = image(blood_splatter_icon)
-		blood_overlay.color = color
-		add_overlay(blood_overlay)
+		var/list/params = GLOB.blood_splatter_icons["[blood_color]"]
+		if(!params)
+			params = layering_filter(icon = icon('icons/effects/blood.dmi', "itemblood"), color = blood_color, blend_mode = BLEND_INSET_OVERLAY)
+			GLOB.blood_splatter_icons["[blood_color]"] = params
+		add_filter("blood_splatter", 1, params)
```

Comment:
If we were to directly just manipulate `filters`, then this would save even more CPU (0.07 vs 0.23) and the memory impact would be something like 0.1 MB.

That said, the filter system we have necessitates keeping a list of filters and it involves copying lists---so the end result is that the vast increase in CPU and memory usage is due to record keeping. Very sad.